artiq/coredevice/phaser.py

```diff
@@ -468,6 +474,32 @@ def get_crc_err(self):
         """
         return self.read8(PHASER_ADDR_CRC_ERR)
 
+    @kernel
+    def measure_frame_timestamp(self):
+        """Perform a register read and record the exact frame timing in `self.frame_tstamp`.
+
+        Deterministic timing requires updates to be schedule at multiples of `self.t_frame` later.
+        See `get_next_frame_timestamp()`.
+        """
+        rtio_output((self.channel_base << 8) | (PHASER_ADDR_BOARD_ID & 0x7f), 0)  # can read any register
+        delay_mu(int64(self.t_frame))
+        self.frame_tstamp = rtio_input_timestamp(rtio_get_counter() + 0xffffff, self.channel_base)
```

Comment:
Please choose a meaningful timeout. Especially since this should be the first thing in `init()` and would be hit first if a device_db is messed up.

Reply:
Changed in `125_000` in 075cb26

Reply:
`rtio_counter` is not the right reference here.
`now_mu() + 10*self.t_frame` or something similar.